typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -54,48 +52,71 @@ async function main() {
     throw new Error(`Invalid chains to skip ${invalidChains}`);
   }
 
-  let failureOccurred = false;
-
   const origins = chains.filter((chain) => !skip || !skip.includes(chain));
-
-  // submit frequently so we don't have to wait a super long time for info to get into the metrics
-  const metricsInterval = setInterval(() => {
-    submitMetrics(metricsRegister, 'kathy', { appendMode: true }).catch(
-      console.error,
-    );
-  }, 1000 * 30);
-
-  for (const origin of origins) {
-    for (const destination of origins.filter((d) => d !== origin)) {
-      const labels = {
-        origin,
-        remote: destination,
-        ...constMetricLabels,
-      };
-      try {
-        await sendMessage(app, origin, destination, gasCalc);
-        messagesSendStatus.labels({ ...labels }).set(1);
-      } catch (err) {
-        console.error(
-          `Error sending message from ${origin} to ${destination}, continuing...`,
-          `${err}`.replaceAll('\n', ' ## '),
-        );
-        failureOccurred = true;
-        messagesSendStatus.labels({ ...labels }).set(0);
-      }
-
-      // Sleep 500ms to avoid race conditions where nonces are reused
-      await sleep(500);
-    }
+  const pairings = diagonalize(
+    origins.map((source) =>
+      origins.map((destination) =>
+        source == destination ? null : { source, destination },
+      ),
+    ),
+  )
+    .filter((v) => v !== null)
+    .map((v) => v!);
```

Comment:
```suggestion
    origins.map((origin) =>
      origins.filter((destination) => origin !== destination).map((destination) => { origin, destination })
      ),
    ),
  );
```

Reply:
We need a square matrix for the diagonlization, so this won't work. This is why I filter after instead. Will update to use origin though.